.github/workflows/build_nuget.yml

```diff
@@ -0,0 +1,104 @@
+name: Windows Nuget
+
+on:
+  push:
+    branches:
+      - master
+      - 'releases/**'
+  pull_request:
+    branches:
+      - '*'
+
+jobs:
+  build_nuget_windows:
+    name: nuget.${{ matrix.toolset }}.${{ matrix.arch_config.arch }}
+    runs-on: ${{matrix.os}}
+    strategy:
+      matrix:
+        os: ["windows-2019"]
+        toolset: ["v141"]
+        arch_config:
+        - { arch: x64, generator_arch: x64}
+  
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          submodules: 'recursive'
+      - uses: ilammy/msvc-dev-cmd@v1
+      - name: Dependencies
+        run: >
+          vcpkg install
+          --host-triplet=${{ matrix.arch_config.arch }}-windows
+          --triplet=${{ matrix.arch_config.arch }}-windows
+          cpprestsdk
+          openssl-windows
+          zlib
+          flatbuffers
+          boost-filesystem
+          boost-thread
+          boost-program-options
+          boost-test
+          boost-align
+          boost-math
+          boost-uuid
+      - name: Configure
+        run: >
+          cmake -S . -B build
+          -G "Visual Studio 16 2019" -A ${{ matrix.arch_config.generator_arch }}
+          -T ${{ matrix.toolset }}
+          -DNATIVE_NUGET_PLATFORM_TAG=${{ matrix.arch_config.arch }}
+          -DBUILD_TESTING=Off
+          -DCMAKE_TOOLCHAIN_FILE=C:/vcpkg/scripts/buildsystems/vcpkg.cmake
+          -DVCPKG_TARGET_TRIPLET=${{ matrix.arch_config.arch }}-windows
+          -DCMAKE_WINDOWS_EXPORT_ALL_SYMBOLS=TRUE
```

Comment:
We should not be relying on `CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS` to export the symbols required for this DLL. It is going to export **significantly** more symbols than is required. It additionally does not enable producing a shared library on Linux or MacOS.

The proper way to do this is as I described in the other comment: https://github.com/VowpalWabbit/reinforcement_learning/pull/471#discussion_r862011640

Reply:
changed to just include the symbols needed for creating a live model, seems like this may be in issue in the future however if other symbols are needed